frontend/types/domains/recruitments.ts

```diff
@@ -0,0 +1,44 @@
+export type RecruitmentStatus = "RECRUITABLE" | "RECRUITING" | "UNRECRUITABLE" | "ENDED";
+
+export type Recruitment = {
+  id: number;
+  title: string;
+  term: {
+    id: number;
+    name: string;
+  };
+  recruitable: boolean;
+  hidden: boolean;
+  startDateTime: string;
+  endDateTime: string;
```

Comment:
r: startDateTime과 endDateTime모두 yyyy-mm-ddThh:mm 형식으로 오는것 같던데 typescript의 template literal type을 이용해서 바꿔볼수 있지 않을까요?
```suggestion
  startDateTime: `${number}-${number}-${number}T${number}:${number}`;
  endDateTime: `${number}-${number}-${number}T${number}:${number}`;
```

이런식으로 사용하면 그냥 string보다 조금 더 제한될것 같네요😎

Reply:
오 이거 좋네요!

Reply:
@soyi47 
소피아~ 상세하게 예시코드까지 작성해주셔서 감사합니다. 

이에 대해서 고민을 해보고 내린 저의 생각입니다. 
예시코드는 타입스크립트를 이용해서 유효성 검사까지 하는 것이라고 생각하는데요. 

저는 두 가지 정도의 이유로 반대합니다. 

- 타입스크립트의 역할
저는 타입스크립트의 역할이 유효성 검사까지라고 생각하지 않습니다. 유효성 검사는 자바스크립트를 이용해서 하는 것이 맞지 않을까 싶네요.

- 컴파일 에러의 이유를 정확히 파악하지 못한다. 
컴파일 에러가 났을 시에, 년월일에서 어떤 값으로 인해서 컴파일 에러가 떴는지 알 수 없고 에러에 대한 상세한 메세지를 처리할 수 없습니다. 

이에 대해서는 어떻게 생각하시나요? 모두의 생각이 궁금하긴 하네요. 

template literal types 관련 아티클 : https://toss.tech/article/template-literal-types
소피아 예시 코디 관련 아티클 : https://blog.logrocket.com/handing-date-strings-typescript/

Reply:
api응답데이터는 컴파일타임에 검증할 수 없어서요 록바 의견에 공감합니당

Reply:
+@ 조금만 더 첨언하면 요 논의는 ts가 런타임에 역할을 할 수 있는지 이야기 되면 좋겠어용

Reply:
요기도 그냥 Date로 관리하는게 어떤지 싶네요 ㅋㅋ

Reply:
@lokba 
록바~ 다시 생각해볼 부분들 제시해주셔서 감사합니다. 👍

제가 '날짜 형식은 어떻게 타이핑 할 수 있을까'에만 집중했던 것 같아요.

말씀하신 부분들을 고려한다면 아래와 같은 방식이 될 것 같습니다.
1. 해당 타입을 string 수준으로만 제한하고
2. 런타임에 유효한 날짜 형식인지, Date로 관리할 수 있는지 확인

그러면 궁금한 부분이 있는데요.

해당 타입은 domain 폴더에 정의되어 있습니다. 
**같은 domain 폴더의 User의 birthday는 type을 Date**로 주고 있고요.
그런 부분들을 고려했을 때는, 어떤 타입을 주는 게 맞다고 생각하시나요?
(저는 domain으로 분류된 타입이라면 date time에 맞게 제한하는 좋을 것 같습니다.)

아마도 기존 로직에 맞춰 타입을 주면서 User의 birthday와 타입이 달라졌을 것 같은데요.
날짜 형식의 값에 대한 상황을 한번 정리하고 가면 좋을 것 같습니다. 👍

Reply:
@soyi47 
날짜 관련한 의견은 해당 코멘트에 남겼습니다! https://github.com/woowacourse/service-apply/pull/612#discussion_r967798867

Reply:
@hwangstar156 
피드백 주신 방향대로 수정하였습니다. 2d869c7 
추후에 날짜와 관련된 값(startDateTime...)들은 Date 타입으로 수정될 것으로 보입니다.